### ISSUE TITLE
shell-provisioner: lots of trouble with a noexec /tmp, failing workaround.

### ISSUE THREAD

**Issue by srenatus** [test-kitchen/test-kitchen#444]
Hi.

Induced by _pingu's requests in freenode/#kitchenci, I had a look at the shell provisioner when /tmp is mounted noexec.  (Guessing that this is the case with SUSE Enterprise Linux.)

Reproducing the issue with Ubuntu is possible, running 

```bash
sudo mount -t tmpfs -o noexec tmpfs /tmp
```

in a Ubuntu VM leads to:

```console
~/tmp$ kitchen converge ubuntu
-----> Starting Kitchen (v1.2.1)
-----> Converging <default-ubuntu-1204>...
       Preparing files for transfer
       Preparing script
       bootstrap.sh not found so Kitchen will run a stubbed script. Is this intended?
       Transfering files to <default-ubuntu-1204>
>>>>>> Converge failed on instance <default-ubuntu-1204>.
>>>>>> Please see .kitchen/logs/default-ubuntu-1204.log for more details
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: SSH exited (1) for command: [sudo -E /tmp/kitchen/bootstrap.sh]
>>>>>> ----------------------
```

The obvious change however, changing the root_path of the provisioner:

```
provisioner:
  name: shell
  root_path: /opt/kitchen
```

fails, too, because of [a bug here](https://github.com/test-kitchen/test-kitchen/blob/master/lib/kitchen/provisioner/shell.rb#L48), missing " around what `sudo` is meant to execute:

```console
~/tmp$ kitchen converge ubuntu
-----> Starting Kitchen (v1.2.1)
-----> Converging <default-ubuntu-1204>...
       Preparing files for transfer
       Preparing script
       bootstrap.sh not found so Kitchen will run a stubbed script. Is this intended?
mkdir: cannot create directory `/opt/kitchen': Permission denied
>>>>>> Converge failed on instance <default-ubuntu-1204>.
>>>>>> Please see .kitchen/logs/default-ubuntu-1204.log for more details
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: SSH exited (1) for command: [sudo -E rm -rf /opt/kitchen/data ; mkdir -p /opt/kitchen]
>>>>>> ----------------------
```

That said, I wonder how we can remedy the noexec issue properly.  I don't know of any platform-independent way (reading /proc? parsing mount(8)?) to check and warn the user.  A robust approach would be to use the proper interpreter, i.e. run "sudo -E sh /tmp/kitchen/bootstrap.sh", but then this should be a configurable setting, allowing for using Python, Ruby, your-favorite-outlandish-shell...

TL;DR: There's a bug in the shell-provisioners init_command ;)

**Comment by smurawski** [comment]
@srenatus Sorry for the delayed respons, but is this still an issue for you?

**Comment by srenatus** [comment]
@smurawski: oh, thanks. No, it's not. Closing... :)
